### DISCUSSION TITLE
What should I do to prepare for possible loss of authenticator access?

### DISCUSSION THREAD

**Discussion by jezcorbett** [community/community#31031] [New to GitHub]
I use google authenticator to access my account, I remember when setting up 2FA something about not being able to recover my account if I lose access to it.

What can I do to protect my ability to access my github account in the event I lose my phone or suchlike?

As an aside, are there any decent alternatives to google authenticator I can use, having to open the app and copy of a 6 digit code that might change while I'm typing it is a bit annoying, if there's a single prompt, verify with thumbprint type authenticator I can use with github, that would be great.

Thanks!

**Comment by melvincwng** [comment]
@jezcorbett, Great question!

If you happen to lose access to your phone that has Google Authenticator on it, please follow these steps as stated in [Google's Help Center](https://support.google.com/accounts/answer/185834?hl=en#zippy=%2Cget-a-new-phone-from-your-carrier%2Crecover-your-account) on how to resolve this issue. The steps basically involve revoking access in your lost phone & how to recover your account's access in another phone/device.

Secondly, I recommend reading this resource on [GitHub Documentation](https://docs.github.com/en/authentication/securing-your-account-with-two-factor-authentication-2fa/configuring-two-factor-authentication) on the various ways to handle 2FA in GitHub. There are many other ways or applications that allow 2FA besides Google Authenticator; you can explore whichever options fits your uses/needs.

Hope this answers your question 😄

**Comment by airtower-luna** [comment]
> I use google authenticator to access my account, I remember when setting up 2FA something about not being able to recover my account if I lose access to it.

The very first thing to do is [download your recovery codes](https://docs.github.com/en/authentication/securing-your-account-with-two-factor-authentication-2fa/configuring-two-factor-authentication-recovery-methods#downloading-your-two-factor-authentication-recovery-codes). Keep them in a safe place. On a backup disk, printout, whatever works for you. Those can get you in to recover your account if your 2FA fails.

> As an aside, are there any decent alternatives to google authenticator I can use, having to open the app and copy of a 6 digit code

The 6 digit codes are key to TOTP, but there are a lot of other TOTP implementations. Notably some of them allow you to make a backup and move it to another device, which is another way to prevent losing your 2FA. For example with [KeePassXC](https://keepassxc.org/) you can move the database file around and that's it. Naturally you should consider carefully which implementation you want to trust.

Additionally you can [set up a security key](https://docs.github.com/en/authentication/securing-your-account-with-two-factor-authentication-2fa/configuring-two-factor-authentication#configuring-two-factor-authentication-using-a-security-key) if your system supports it.

**Comment by jezcorbett** [comment]
Thanks for the answers, everyone! This is very helpful indeed.
